### PR TITLE
chore: replace jsonwebtoken with jwt-compact

### PR DESCRIPTION
**Pull request by Finistere**
jsonwebtoken doesn't compile to wasm locally because of ring (who knows
why it works in the CI).
For the v1 auth we rely on jwt-compact anyway and WASM-compatbility is
an explicit goal.

# Description

Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.

# Type of change

- [ ] 💔 Breaking
- [ ] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

**Review by tomhoule — APPROVED** [review]

File: engine/crates/gateway-v2/auth/src/jwt.rs [hunk 1]
```diff
@@ -82,93 +120,122 @@ impl Authorizer for JwtProvider {
...
-        for key in decoding_keys(&metadata.jwks, &jose_header) {
-            if let Ok(TokenData { claims, .. }) = jsonwebtoken::decode(token_str, &key, &validation) {
-                use base64::{engine::general_purpose::URL_SAFE_NO_PAD, Engine as _};
-                let signature = URL_SAFE_NO_PAD
-                    .decode(token_str.rsplit('.').next().expect("valid jwt"))
-                    .expect("valid jwt");
-                return Some(AccessToken::Jwt(JwtToken { claims, signature }));
+        if let Some(expected) = self.config.jwks.audience.as_ref() {
+            let audience = token.claims().custom.audience.as_ref()?;
+            if audience.iter().all(|aud| aud != expected) {
```
Comment (obmarg) [inline engine/crates/gateway-v2/auth/src/jwt.rs:156]: `all` on an empty iterator returns true - I don't think that's what you want here?

Comment (Finistere) [reply engine/crates/gateway-v2/auth/src/jwt.rs:156]: damn, good catch, just at the moment I'm merging xD
thanks!

Comment (Finistere) [reply engine/crates/gateway-v2/auth/src/jwt.rs:156]: ah no that's what I want. I do test it. If there is nothing, it's true and I'm returning `None`

Comment (Finistere) [reply engine/crates/gateway-v2/auth/src/jwt.rs:156]: not the clearest, have to admit it

Comment (obmarg) [reply engine/crates/gateway-v2/auth/src/jwt.rs:156]: Oh shit yes, cool.  Ignore me then.

**Pull request merged**